doc/source/train/examples/pytorch/distributing-pytorch/README.ipynb

```diff
@@ -0,0 +1,599 @@
+    "            loss.backward()\n",
+    "            optimizer.step()\n",
+    "\n",
+    "        model.eval()\n",
+    "        test_loss, num_correct, num_total, num_batches = 0, 0, 0, 0\n",
+    "        with torch.no_grad():\n",
+    "            for batch in tqdm(test_dataloader, desc=f\"Test Epoch {epoch}\"):\n",
+    "                # [2] Each Ray Data batch is a dict so you must access the\n",
+    "                # underlying data using the appropriate keys.\n",
+    "                # =======================================================\n",
+    "                X, y = batch['image'], batch['label']\n",
+    "                pred = model(X)\n",
+    "                loss = loss_fn(pred, y)\n",
+    "\n",
+    "                test_loss += loss.item()\n",
+    "                num_total += y.shape[0]\n",
+    "                num_batches += 1\n",
+    "                num_correct += (pred.argmax(1) == y).sum().item()\n",
+    "\n",
+    "        test_loss /= num_batches\n",
+    "        accuracy = num_correct / num_total\n",
+    "\n",
+    "        with tempfile.TemporaryDirectory() as temp_checkpoint_dir:\n",
+    "            torch.save(\n",
+    "                model.module.state_dict(),\n",
+    "                os.path.join(temp_checkpoint_dir, \"model.pt\")\n",
+    "            )\n",
+    "            ray.train.report(\n",
+    "                metrics={\"loss\": test_loss, \"accuracy\": accuracy},\n",
+    "                checkpoint=ray.train.Checkpoint.from_directory(temp_checkpoint_dir),\n",
+    "            )\n",
+    "            if ray.train.get_context().get_world_rank() == 0:\n",
+    "                print({\"epoch_num\": epoch, \"loss\": test_loss, \"accuracy\": accuracy})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, run the training function with the Ray Data Dataset on the Ray Cluster with 8 GPU workers."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def train_cifar_10(num_workers, use_gpu):\n",
+    "    global_batch_size = 512\n",
+    "\n",
+    "    train_config = {\n",
+    "        \"lr\": 1e-3,\n",
+    "        \"epochs\": 1,\n",
+    "        \"batch_size_per_worker\": global_batch_size // num_workers,\n",
+    "    }\n",
+    "\n",
+    "    # Configure computation resources\n",
+    "    scaling_config = ScalingConfig(num_workers=num_workers, use_gpu=use_gpu)\n",
+    "    run_config = RunConfig(storage_path=\"/mnt/cluster_storage\", name=\"ray_train_run\")\n",
+    "\n",
+    "    # Initialize a Ray TorchTrainer\n",
+    "    trainer = TorchTrainer(\n",
+    "        train_loop_per_worker=train_func_per_worker,\n",
+    "        # [1] With Ray Data you pass the Dataset directly to the Trainer.\n",
+    "        # ==============================================================\n",
+    "        datasets={\"train\": train_dataset, \"test\": test_dataset},\n",
```

Comment:
nit: "valid" set rather than "test" set

Reply:
Changed all references to "test" in the train_func to "valid" but kept generic loading utilities as "test" since they could have been used for testing. Lmk if this is fine.